tests/src/JIT/Performance/CodeQuality/Span/Indexer.cs

```diff
@@ -0,0 +1,925 @@
+                return result;
+            },
+            "KnownSizeCtor({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte TestKnownSizeCtor(byte[] a, int innerIterationCount)
+        {
+            Span<byte> data = new Span<byte>(a, 0, 1024);
+            byte x = 0;
+
+            for (int i = 0; i < innerIterationCount; i++)
+            {
+                x = 0;
+                for (var idx = 0; idx < data.Length; idx++)
+                {
+                    x ^= data[idx];
+                }
+            }
+
+            return x;
+        }
+
+        [Benchmark(InnerIterationCount = Iterations)]
+        [InlineData(DefaultLength)]
+        [Category("Span known size bounds check elimination")]
+        public static void KnownSizeCtor2(int length)
+        {
+            if (length < 1024)
+            {
+                throw new Exception("test requires at least 1024 byte length");
+            }
+
+            byte[] a = GetData(length);
+
+            Invoke((int innerIterationCount) =>
+            {
+                byte result = TestKnownSizeCtor2(a, innerIterationCount);
+                return result;
+            },
+            "KnownSizeCtor2({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte TestKnownSizeCtor2(byte[] a, int innerIterationCount)
+        {
+            Span<byte> data1 = new Span<byte>(a, 0, 512);
+            Span<byte> data2 = new Span<byte>(a, 512, 512);
+            byte x = 0;
+
+            for (int i = 0; i < innerIterationCount; i++)
+            {
+                x = 0;
+                for (var idx = 0; idx < data1.Length; idx++)
+                {
+                    x ^= data1[idx];
+                }
+                for (var idx = 0; idx < data2.Length; idx++)
+                {
+                    x ^= data2[idx];
+                }
+            }
+
+            return x;
+        }
+
+        [Benchmark(InnerIterationCount = Iterations)]
+        [InlineData(DefaultLength)]
+        [Category("Same index in-loop redundant bounds check elimination")]
+        public static void SameIndex1(int length)
+        {
+            byte[] a = GetData(length);
+
+            Invoke((int innerIterationCount) =>
+            {
+                Span<byte> s = new Span<byte>(a);
+                byte result = 0;
+                for (int i = 0; i < innerIterationCount; ++i)
+                {
+                    result = TestSameIndex1(s);
+                }
+                return result;
+            },
+            "SameIndex1({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte TestSameIndex1(Span<byte> data)
+        {
+            byte x = 0;
+            byte y = 0;
+
+            for (var idx = 0; idx < data.Length; idx++)
+            {
+                x ^= data[idx];
+                y ^= data[idx];
+            }
```

Comment:
Good point, will update the loop bounds on those tests to be argument values.

On `WriteViaIndexer2` I could go back to 0 lower bounds. It started off as a clone of `WriteViaIndexer1`...